modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/featureflag/FirebaseRemoteFeatureInitializer.kt

```diff
@@ -0,0 +1,38 @@
+package au.com.shiftyjelly.pocketcasts.utils.featureflag
+
+import au.com.shiftyjelly.pocketcasts.helper.BuildConfig
+import com.google.firebase.remoteconfig.FirebaseRemoteConfig
+import java.util.concurrent.atomic.AtomicBoolean
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.withTimeout
+import timber.log.Timber
+
+@Singleton
+class FirebaseRemoteFeatureInitializer @Inject constructor(
+    private val firebaseRemoteConfig: FirebaseRemoteConfig,
+) {
+
+    private companion object {
+        const val DEFAULT_TIMEOUT = 3_000L
+    }
+
+    private val isInitialized = AtomicBoolean(BuildConfig.DEBUG || BuildConfig.IS_PROTOTYPE)
```

Comment:
❓ I'm not sure I understand why `AtomicBoolean` is used here. Where its atomicity is being used?